bgpd/bgpd.c

```diff
@@ -1345,6 +1345,7 @@ struct peer *peer_new(struct bgp *bgp)
 	peer->bgp = bgp_lock(bgp);
 	peer = peer_lock(peer); /* initial reference */
 	peer->password = NULL;
+	peer->max_packet_size = BGP_MAX_EXTENDED_MESSAGE_PACKET_SIZE;
```

Comment:
Shouldn't this be assigned based on whether extended message support is configured/exchanged with peer or not ?

Reply:
It's kinda tricky. The main issue is that this is set on peer create, but the size of the message is negotiated during the OPEN message and can't be reallocated according to that. Reallocating ring buffers, stopping/starting I/O on the fly sounds dangerous. So, having a 65k in advance sounds good to me. Outgoing buffers allocated according to if capability exchanged or not.

Reply:
Don't know the complete work flow for ring buffers allocation/free, so can't comment much on that, but this essentially defeat the purpose of this variable if we always allocate 65K anyway.

Reply:
Frankly speaking, I don't see a bad things having a maximum buffer size here. This is almost the same as setting maximum ring buffers by default for a NIC... Which is normal. I agree about the stack size, I see @rzalamena fixed that already.